nomad/grapl-core.nomad

```diff
@@ -0,0 +1,428 @@
+# For rust logging output
+variable "RUST_LOG" {
+    type = string
+    default = "INFO"
+}
+
+# Which tag to use for grapl-specific images
+variable "tag" {
+    type = string
+    default = "latest"
+}
+
+# The container registry that we can find grapl-specific services
+variable "container_registry" {
+    type = string
+    default = "localhost:5000"
+}
+
+variable "aws_region" {
+    type = string
+    default = "us-west-2"
+}
+
+variable "aws_sqs_url" {
+    type = string
+}
+
+variable "aws_account_id" {
+    type = string
+    default = "000000000000"
+}
+
+variable "dgraph_replicas" {
+    type = number
+    default = 1
+}
+
+variable "dgraph_shards" {
+    type = number
+    default = 1
+}
+
+# Where is redis located?
+variable "redis_endpoint" {
+    type = string
+}
+
+# What is the name of the schema table?
+variable "schema_table_name" {
+    type = string
+}
+
+# https://github.com/grapl-security/grapl/blob/af6f2c197d52e9941047aab813c30d2cbfd54523/pulumi/infra/dynamodb.py#L118
+# Without deployment name
+variable "session_table" {
+    type = string
+    default = "dynamic_session_table"
+}
+
+# How many graph mergers should be running in tandem
+variable "graph_mergers" {
+    type = number
+    default = 1
+}
+
+# How many node identifiers should be running in tandem
+variable "node_identifiers" {
+    type = number
+    default = 1
+}
+
+variable "node_identifiers_retry" {
+    type = number
+    default = 1
+}
+
+locals {
+    # DGraph Alphas (shards * replicas)
+    dgraph_alphas = [for alpha_id in range(0, var.dgraph_replicas * var.dgraph_shards): {
+        id: alpha_id,
+        grpc_private_port: alpha_id + 7080,
+        grpc_public_port: alpha_id + 9080,
+        http_port: alpha_id + 8080
+    }]
+
+    # DGraph Zeros (replicas)
+    dgraph_zeros = [for zero_id in range(1, var.dgraph_replicas): {
+        id: zero_id,
+        grpc_private_port: zero_id + 5080,
+    }]
+
+    # String that contains all of the Zeros for the Alphas to talk to and ensure they don't go down when one dies
+    zero_alpha_connect_str = join(",", [for zero_id in range(0, var.dgraph_replicas): "localhost:${zero_id + 5080}"])
+
+    # String that contains all of the running Alphas for clients connecting to Dgraph (so they can do loadbalancing)
+    alpha_grpc_connect_str = join(",", [for alpha in local.dgraph_alphas: "localhost:${alpha.grpc_public_port}"])
```

Comment:
Is this `localhost` here specifically limiting this file to being used just on our workstations, or is this somehow mapped via Consul / DNS / Hashimagic to be used as-is in real multi-host deployments?

Reply:
> Is this localhost here specifically limiting this file to being used just on our workstations, or is this somehow mapped via Consul / DNS / Hashimagic to be used as-is in real multi-host deployments?

This is for consul connect. In consul connect land, all of the remote systems appear as "localhost" (see: my grapl hashistack presentation)

Reply:
Nice